x/clp/keeper/msg_server.go

```diff
@@ -109,38 +105,43 @@ func (k msgServer) Swap(goCtx context.Context, msg *types.MsgSwap) (*types.MsgSw
 	var (
 		priceImpact sdk.Uint
 	)
-	if !(k.tokenRegistryKeeper.IsDenomWhitelisted(ctx, msg.ReceivedAsset.Symbol) || (k.tokenRegistryKeeper.IsDenomWhitelisted(ctx, msg.SentAsset.Symbol))) {
+	registry := k.tokenRegistryKeeper.GetRegistry(ctx)
+	sAsset, err := k.tokenRegistryKeeper.GetEntry(registry, msg.SentAsset.Symbol)
+	if err != nil {
 		return nil, types.ErrTokenNotSupported
 	}
-	if !k.tokenRegistryKeeper.CheckDenomPermissions(ctx, msg.ReceivedAsset.Symbol, types.GetCLPermissons()) || !k.tokenRegistryKeeper.CheckDenomPermissions(ctx, msg.SentAsset.Symbol, types.GetCLPermissons()) {
+	rAsset, err := k.tokenRegistryKeeper.GetEntry(registry, msg.ReceivedAsset.Symbol)
+	if err != nil {
 		return nil, types.ErrTokenNotSupported
 	}
+	if !k.tokenRegistryKeeper.CheckEntryPermissions(sAsset, []tokenregistrytypes.Permission{tokenregistrytypes.Permission_CLP}) {
```

Comment:
NIT : The CLP permissions were contained in a function , not sure why that was changed .